### PR TITLE
Cleaning, Automatic Control, and Bezier Curves

### PR DESCRIPTION
**Pull request by ammrat13**
* Cleaned up old files. Deleted `racecar/` and `tests`. Renamed `TrainingBot/` to make styling consistent.
* Created automatic control system. Added logic that allows humans to take control at any time by pressing any key. Created a `generate_target_velocities()` generator for automatic control.
* Implemented the kinematics equations from @maxrudolph1. Used them to implement a single integrator on _θ_, _x_, and _y_, as well as generating our target angular velocities for the wheels. Also implemented guidance equations for following a Bezier curve by dead-reckoning.

### CODE REVIEW

File: simulator/game.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 import time
 import pybullet as p
 
-from simulator.racecar_agent import RacecarAgent
 from simulator.field import Field
 from simulator.legos import Legos
 from simulator.utilities import Utilities
```
Comment (binitshah) [inline simulator/game.py]: **TODO:** Remove all changes to `game.py` in this PR

File: simulator/trainingbot_agent.py [hunk 2]
```diff
@@ -84,4 +104,86 @@ def update(self):
...
+                        p.getBasePositionAndOrientation(self.robot)[0],
+                        p.getBasePositionAndOrientation(self.robot)[1],
+                        (0,0,L + 0.074676),
+                        (0,0,0,1))[0]
+
+        lid = p.addUserDebugLine(
+                        [BEZIERX[0]+BEZIERX[1]*currentU+BEZIERX[2]*currentU**2+BEZIERX[3]*currentU**3,
+                         BEZIERY[0]+BEZIERY[1]*currentU+BEZIERY[2]*currentU**2+BEZIERY[3]*currentU**3,
+                         -10],
+                        [BEZIERX[0]+BEZIERX[1]*currentU+BEZIERX[2]*currentU**2+BEZIERX[3]*currentU**3,
```
Comment (binitshah) [inline simulator/trainingbot_agent.py]: **TODO:** Move all of this work to [Planning2020](https://github.com/GT-IEEE-Robotics/Planning2020) repo.

File: simulator/utilities.py [hunk 3]
```diff
@@ -39,3 +39,25 @@ def add_debug_pose(position=np.array([0,0,0]),
         py_uid = p.addUserDebugLine(po, py, y_color, lineWidth, lifeTime, parentObjectUniqueId, parentLinkIndex, replaceItemUniqueIds[1])
         pz_uid = p.addUserDebugLine(po, pz, z_color, lineWidth, lifeTime, parentObjectUniqueId, parentLinkIndex, replaceItemUniqueIds[2])
         return (px_uid, py_uid, pz_uid)
+
+    def add_bezier_curve(BEZIERX,
```
Comment (binitshah) [inline simulator/utilities.py]: Follow naming scheme of above utility methods

**Review by binitshah — CHANGES_REQUESTED** [review]
Additional Todos:
 - Rebase this your fork from upstream and update this PR
 - Don't delete `test\` or racecar

**Comment by ammrat13** [comment]
I think I've fixed it.

File: simulator/trainingbot_agent.py [hunk 4]
```diff
@@ -22,7 +24,10 @@ def __init__(self, motion_delta=0.5, skew=0.0):
         self.max_force = 1
         self.motion_delta = motion_delta
         self.velocity_limit = 5
+        # As fractions of self.velocity_limit
+        # Range from -1.0 to 1.0
```
Comment (binitshah) [inline simulator/trainingbot_agent.py]: Does not range [-1.0, 1.0]. It is [-self.velocity_limit, self.velocity_limit]

File: simulator/trainingbot_agent.py [hunk 5]
```diff
@@ -5,6 +5,8 @@
 Last Modified: Binit on 2/20
 """
 
+from math import cos, sin, sqrt
```
Comment (binitshah) [inline simulator/trainingbot_agent.py]: Remove imports

**Review by binitshah — CHANGES_REQUESTED** [review]

File: simulator/trainingbot_agent.py (hunk 4 above)
Comment (ammrat13) [reply simulator/trainingbot_agent.py]: Fixed it in the code, forgot to remove the comment

**Comment by ammrat13** [comment]
Think I've fixed it

**Comment by binitshah** [comment]
Closing this PR. We really want to plot trajectories, not just bezier curves. I'll take care of this. We can move the bezier visualizer to Planning as a utility function.

Sorry about the back and forth on this PR.

**Pull request closed without merging**